lib/api/transactions.js

```diff
@@ -306,48 +307,37 @@ module.exports = function (app) {
 
 			const t = new Transaction();
 
-			if (transaction.votes) {
-				return async.waterfall([
-					(cb) => {
-						if (transaction.votes.added.length) {
-							const added = [];
-							_.each(transaction.votes.added, (vote) => {
-								added.push({ publicKey: vote });
-							});
-
-							return async.forEach(added, (add, callback) => {
-								t.getDelegate(add, callback);
-							}, () => {
-								transaction.votes.added = added;
-								return cb(null, transaction);
-							});
+			if (transaction && transaction.asset && transaction.asset.votes) {
```

Comment:
I would move this entire block to a separated function as part of the async.waterfall